src/engine/CDSmtCore.cpp

```diff
@@ -2,7 +2,7 @@
 /*! \file CDSmtCore.cpp
```

Comment:
There are no unit tests for this class. I did my best effort to integrate the SoI change to this. But we probably need to revisit testing the correctness of this class later.